plugins/generator-1.19.2/forge-1.19.2/templates/gui/gui_window.java.ftl

```diff
@@ -203,6 +203,28 @@ public class ${name}Screen extends AbstractContainerScreen<${name}Menu> {
 				<#assign btid +=1>
 		</#list>
 
+		<#list data.getComponentsOfType("ImageButton") as component>
+				this.addRenderableWidget(new ImageButton(this.leftPos + ${(component.x - mx/2)?int}, this.topPos + ${(component.y - my/2)?int},
+					${component.width}, ${component.height}, 0, 0, ${component.height}, new ResourceLocation("${modid}:textures/screens/atlas/${component.getName()}.png"), ${component.width}, ${component.height * 2}, e -> {
+							<#if hasProcedure(component.onClick)>

```

Comment:
Code below this is same for button and image button, consider making eg. macro for this code part so it is not duplicated